enterprise_access/apps/subsidy_access_policy/models.py

```diff
@@ -803,7 +879,18 @@ def can_redeem(self, lms_user_id, content_key, skip_customer_user_check=False):
         # learner can redeem the subsidy access policy
         return (True, None, existing_redemptions)
 
-    def credit_available(self, lms_user_id=None):
+    def credit_available(self, lms_user_id, skip_customer_user_check=False):
+        """
+        Determine whether a learner has credit available for the subsidy access policy.
+        """
+        is_credit_available = super().credit_available(lms_user_id, skip_customer_user_check)
+        if not is_credit_available:
+            return False
+
+        if self.per_learner_spend_limit is None:
+            return True
+
+        # Validate whether learner has enough remaining balance (spend) for this policy.
```

Comment:
I think `self.remaining_balance_per_user()` needs to account for negative aggregates before comparing to the per-learner spend limit.

Reply:
@iloveagent57 Good catch! I've updated to account for negative aggregates here and the associated tests in the most recent commit.